docs/development/design/exit-node.md

```diff
@@ -0,0 +1,145 @@
+
+An example workflow of an agent node looks like the following (open to adding client to the command):
+
+```shell
+# exit node reflexive address = x.227.102.183
+# exit node client reflexive address = x.31.152.254
+
+# Before enabling exit-node on the client
+$ wget -qO - icanhazip.com
+x.31.152.254
+
+$ sudo nexctl nexd exit-node enable
+Successfully enabled exit node client on this device
+
+$ sudo nexctl nexd exit-node list
+ENDPOINT ADDRESS         PUBLIC KEY
+x.227.102.183:44728     6/CwH/gzz9jdKnxeVpWFivycFQMBkniLeBNFQq0+f04=
+
+$ wget -qO - icanhazip.com
+x.227.102.183
+
+$ sudo nexctl nexd exit-node disable
+Successfully disabled exit node client on this device
+
+# After subscribing to the exit-node, the client is now originating traffic from the exit node.
+$ wget -qO - icanhazip.com
+x.31.152.254
+```
+
+A diagram of the traffic flow is as follows:
+
+```mermaid
+graph LR
+subgraph Nexodus Mesh with All Devices Peered to one another
+Nex_Device_4["Nexd-1 (100.100.0.1)
+exit-node client"]
+Nex_Device_5["Nexd-2 (100.100.0.2)
+exit-node client"]
+Nex_Device_6["Nexd-3 (100.100.0.3)
+exit-node client"]
+Nexodus_Network_Router_B["Nexd-4 Exit Node 
+wg0 100.100.0.4
+eth(n) 172.16.100.10/24"]
+end
+
+subgraph External Network 
+Internet["Network or Internet
+Receives Traffic from
+the Nexodus Network
+as sourced from eth(n)
+172.16.100.10 via SNAT"]
+end
+
+ Nexodus_Network_Router_B-- "eth(n)" --> Internet
+Nex_Device_4 -- wg0 --> Nexodus_Network_Router_B
+Nex_Device_5 -- wg0 --> Nexodus_Network_Router_B
+Nex_Device_6 -- wg0 --> Nexodus_Network_Router_B
+```
+
+### Client Out of Band Services
+
+- Since all traffic is being tunneled there is a high likelihood that at some point the device will lose connectivity to the exit node and become orphaned. Because of that, we are baking in as many safeguards as possible and options to recover connectivity.
+
+1. Nexodus services will travel out of band via the device's interface containing a default route via Linux policy routing. A new routing table is created with a default route pointing to the same default route in the default namespace. Then using combinations of nftables and rules in the Linux RPDB (routing policy database) traffic destined for the OOB services is mangled and forwarded bypassing the Nexodus driver interface (for example wg0). The initial out of band services are:
+   - HTTPS port 443 to the API server's destination IP(s). If an exit node becomes unavailable, the exit node client device would no longer be able to reach the device. This would orphan the node and require a user to run nexctl to disable the exit node subscription. To allow an administrator or user of many devices to be able to dis-enroll a device from the web UI, we need to maintain api-server connectivity even if an exit node is unavailable.
```

Comment:
Instead of port 443, I would specify it as the API server port. `nexd` will know if it's on a nonstandard port since it has to be started with the URL to the API server.

Reply:
```suggestion
   - HTTPS port 443 to the API server's destination IP(s). If an exit node becomes unavailable, the exit node client device would no longer be able to reach the device. This would orphan the node and require a user to run nexctl to disable the exit node subscription. To allow an administrator or user of many devices to be able to dis-enroll a device from the web UI, we need to maintain API server connectivity even if an exit node is unavailable.
```